server/src/test/java/org/bonitasoft/console/common/server/servlet/IconServletTest.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (C) 2016 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This library is free software; you can redistribute it and/or modify it under the terms
+ * of the GNU Lesser General Public License as published by the Free Software Foundation
+ * version 2.1 of the License.
+ * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+ * Floor, Boston, MA 02110-1301, USA.
+ **/
+
+package org.bonitasoft.console.common.server.servlet;
+
+import static org.mockito.Mockito.*;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.bonitasoft.engine.api.IdentityAPI;
+import org.bonitasoft.engine.exception.NotFoundException;
+import org.bonitasoft.engine.identity.impl.IconImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/**
+ * @author Baptiste Mesta
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class IconServletTest {
```

Comment:
Please see comments below.
No offense but main problem here is that you are testing code and not behavior.

We can see that because tests are not well named and all expectations are done with `verify`. You don't want to verify that a method has been called but expected behavior has been realized.